diff --git a/libhb/encavcodec.c b/libhb/encavcodec.c
index 40a9162fe..74e09ffc2 100644
--- a/libhb/encavcodec.c
+++ b/libhb/encavcodec.c
@@ -625,13 +625,6 @@ int encavcodecInit( hb_work_object_t * w, hb_job_t * job )
 
     }
 
-    if (job->vcodec == HB_VCODEC_FFMPEG_MF_H265)
-    {
-        // The Qualcomm HEVC is buggy with bf=1; bf=2 is the recommended
-        // setting.
-        context->max_b_frames = 2;
-    }
-
     if (job->vcodec == HB_VCODEC_FFMPEG_MF_H264 ||
         job->vcodec == HB_VCODEC_FFMPEG_MF_H265)
     {
